### PR TITLE
Fix PR 282

### PR DESCRIPTION
**Pull request by lightwalker-eth**

### CODE REVIEW

File: packages/nameguard-react/src/components/ReportIcon/index.tsx [hunk 1]
```diff
@@ -63,57 +46,46 @@ export function ReportIcon({
...
+    tooltipTitle = "Error loading report";
+  } else if (!data) {
+    icon = <RatingLoadingIcon size={size} {...props} />; // TODO: add isInteractive={true} prop?
+    tooltipIcon = (
+      <RatingLoadingIcon
+        size={RatingIconSize.small}
+        fill="#dddddd"
+        {...props}
       />
+    ); // TODO: why is it necessary to pass this `fill` prop to make this visible? That's a bad design. It should work properly without needing to pass that.
```
Comment (FrancoAguzzi) [inline packages/nameguard-react/src/components/ReportIcon/index.tsx:71]: Hi @lightwalker-eth, I have already spent an hour on this today and I remember to have invested in the past on this strange need. In fact it is not nice to have to set `fill` in here but curiously this is needed for the Tooltip version of `RatingLoadingIcon`... Maybe it is good to come back to this after merging your PR?

File: packages/nameguard-react/src/components/ReportIcon/index.tsx [hunk 2]
```diff
@@ -126,62 +98,51 @@ export function ReportIcon({
...
+        /*
+          TODO: Is there a problem making this a <div? Advice appreciated.
+          Whatever you suggest, please be sure to test clicking on it on
+          `/docs/report`.
+
+          I also added `cursor-pointer` to this div to ensure the hover state UX is correct.
+          I tried changing this div to a button, but then started getting errors.
+        */
+        <div
+          onClick={() => openReport(ensName, onOpenReport)}
```
[the remark this answers is not in the document]
Comment (FrancoAguzzi) [reply packages/nameguard-react/src/components/ReportIcon/index.tsx:113]: @lightwalker-eth no problem in making this a div if we specify to web crawlers that it behaves as a button.

File: packages/nameguard-react/src/components/ReportIcon/index.tsx [hunk 3]
```diff
@@ -63,57 +46,46 @@ export function ReportIcon({
...
-        onTooltipClickOverride={(e?: React.MouseEvent) => {
-          if (e) e.stopPropagation();
-          onClickHandler(ClickHandlerFor.tooltip);
-        }}
+    icon = <RatingUnknownIcon size={size} isInteractive={true} {...props} />;
+    tooltipIcon = <RatingUnknownIcon size={RatingIconSize.small} />;
+    tooltipTitleClass = "font-semibold mb-1 text-white";
+    tooltipTitle = "Error loading report";
+  } else if (!data) {
+    icon = <RatingLoadingIcon size={size} {...props} />; // TODO: add isInteractive={true} prop?
```
[the remark this answers is not in the document]
Comment (lightwalker-eth) [reply packages/nameguard-react/src/components/ReportIcon/index.tsx:64]: I think we should add this, but it's not a priority for now so I added a story to track this goal for the future: https://app.shortcut.com/ps-web3/story/25745/add-hover-state-for-loading-state-of-ratingicon

File: packages/nameguard-react/src/components/ReportIcon/index.tsx (hunk 1 above)
Comment (lightwalker-eth) [reply packages/nameguard-react/src/components/ReportIcon/index.tsx:71]: @FrancoAguzzi Agreed, I created a separate story to track this goal here: https://app.shortcut.com/ps-web3/story/25745/refine-loading-state-of-ratingicon

File: packages/nameguard-react/src/components/ReportBadge/index.tsx [hunk 4]
```diff
@@ -25,131 +23,49 @@ interface ReportBadgeProps {
...
-        onClickHandler(ClickHandlerFor.badge);
-      }}
+    <button onClick={() => openReport(ensName, onOpenReport)}
+      className={
+        "flex-shrink-0 appearance-none bg-white transition-colors hover:bg-gray-50 border border-gray-200 rounded-md px-2.5 py-1.5 inline-flex items-center"
+      }
     >
       <DisplayedName
-        textStylingClasses="cursor-pointer pr-1.5"
+        textStylingClasses="cursor-pointer pr-1.5" // TODO: It's interesting that "cursor-pointer" is needed for the case that the `TruncatedText` for the `DisplayedName` overflows. Does this reflect an opportunity to refine `DisplayedName` or `TruncatedText`?
```
[the remark this answers is not in the document]
Comment (lightwalker-eth) [reply packages/nameguard-react/src/components/ReportBadge/index.tsx:54]: Cool. I've updated the comments here based on your advice.

File: packages/nameguard-react/src/components/ReportIcon/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (lightwalker-eth) [reply packages/nameguard-react/src/components/ReportIcon/index.tsx:113]: Thanks for suggestion. Updated code :+1:

File: packages/nameguard-react/src/components/ReportIcon/index.tsx (hunk 1 above)
Comment (FrancoAguzzi) [reply packages/nameguard-react/src/components/ReportIcon/index.tsx]: Added comments to this story so we further define this for future coding! 🙂

File: packages/nameguard-react/src/components/ReportIcon/index.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (FrancoAguzzi) [reply packages/nameguard-react/src/components/ReportIcon/index.tsx]: Added comments to this story so we further define this for future coding! 🙂

File: packages/nameguard-react/src/components/ReportIcon/index.tsx [hunk 5]
```diff
@@ -63,127 +78,98 @@ export function ReportIcon({
...
     }
-  };
+  }, [data]);
+
+  let icon: React.ReactNode;
+  let tooltipIcon: React.ReactNode;
+  let tooltipTitleClass: string;
+  let tooltipTitle: string;
+  let tooltipSubtitle: string | undefined = undefined;
+  let tooltipMessage: string | undefined = undefined;
```
Comment (FrancoAguzzi) [inline packages/nameguard-react/src/components/ReportIcon/index.tsx:96]: This is a synonym of:

```suggestion
  let tooltipSubtitle: string | undefined;
  let tooltipMessage: string | undefined;
```

Comment (FrancoAguzzi) [reply packages/nameguard-react/src/components/ReportIcon/index.tsx:96]: Feel free to choose one between two

**Pull request merged**